### DISCUSSION TITLE
How to change the the color of all the components of MaterialFx

### DISCUSSION THREAD

**Discussion by shah-dhwanil** [palexdev/MaterialFX#151] [Archived]
Great work on MaterialFX. Loved it.
How to change the overall purple theme of MaterialFx to some another color?

**Comment by palexdev** [comment]
It's easy, just customize the controls with CSS
Like for any other JavaFX component

**Reply by shah-dhwanil** [reply]
But I will have do it for all components or is there any css for all components

**Reply by palexdev** [reply]
It depends, are you familiar with JavaFX?
If not I suggest you looking at some tutorials

Each component has it's own stylesheet
But, say you add all the components in a "root" pane/node
At that point you can style all the components you added in a single stylesheet by adding it to the "root"

To check how can you style my components I suggest you looking at the original CSS files it's pretty easy to understand them, there's also the wiki

**Comment by megabyte6** [comment]
If I wanted a dark theme, would I have to do something like this for every style [here](https://github.com/palexdev/MaterialFX/blob/main/materialfx/src/main/resources/io/github/palexdev/materialfx/css/MFXScrollPane.css) that has `-fx-background-color`?
![image](https://user-images.githubusercontent.com/72172597/229269711-3d0d9f53-cd0e-45f4-9fb6-91cfb9e4680d.png)

**Reply by palexdev** [reply]
Yes, unfortunately I didn't build MaterialFX with theming in mind, I thought that adding user stylesheets was gonna be just enough.
The [rewrite](https://github.com/palexdev/MaterialFX/tree/rewrite) will be much better on this, but it will require a lot of time before the first usable version

**Reply by megabyte6** [reply]
Ah, I see. Thank you. Looking forward to the rewrite! 👍
